web/client/utils/FeatureInfoUtils.js

```diff
@@ -32,7 +32,7 @@ const regexpBody = /^[\s\S]*<body[^>]*>([\s\S]*)<\/body>[\s\S]*$/i;
 const regexpStyle = /(<style[\s\=\w\/\"]*>[^<]*<\/style>)/i;
 
 export function parseHTMLResponse(res) {
-    if ( typeof res.response === "string" && res.response.indexOf("<?xml") !== 0 ) {
+    if ( typeof res.response === "string" && (res.response.indexOf("<?xml") !== -1 || res.response.indexOf("<?xml") !== 0) ) {
```

Comment:
With this new statement the part `(res.response.indexOf("<?xml") !== -1 || res.response.indexOf("<?xml") !== 0)` will return true for each index position so it has not a real effect.

from @offtherailz 
> @MuhweziDeo The linked interceptOGCError is only an example, but it is not buggy.
> I think for the moment you can proceed improving the check in featureInfoUtils to do a real check (strings starting with <?xml are still valid html, an error should intercept the <ExceptionReport tag. To do a test about the error response you can try to do the same http request with an invalid format.

Before we used to check the initial xml definition to ensure we were parsing an error or the actual valid response but this is not valid for all the servers.
We know that OGC error uses the 200 status and it should contain the Exception tag, we have an example from the https://github.com/geosolutions-it/MapStore2/blob/master/web/client/utils/ObservableUtils.js#L35. (from the issue description).

I tried to generate an error in their instance by changing the `layers` field with an invalid layer and it seems their service return a ServiceException tag
![image](https://user-images.githubusercontent.com/19175505/109962968-27652c00-7cec-11eb-86a4-1c1badb0b55d.png)

We could apply this statement "is a string, does not contain <ExceptionReport and does not contain <ServiceExceptionReport tags"
- `typeof res.response === "string" && res.response.indexOf("<ExceptionReport") === -1 && res.response.indexOf("<ServiceExceptionReport") === -1`

What do you think @offtherailz ?

Reply:
@allyoucanmap this makes sense but what are the chances that there isnt other exceptions that can be returned

Reply:
@MuhweziDeo, I talked with @offtherailz and the xml check is an old control so we should remove it.
So we should keep this one:
`if ( typeof res.response === "string") {`

We need 2 unit tests one with an error response and one with a valid html.

Here an error example:
```xml
<?xml version='1.0' encoding="ISO-8859-1"  standalone="no" ?>
<ServiceExceptionReport version="1.1.1">
  <ServiceException code="InvalidFormat">
    <![CDATA[
    Parámetros erroneos:
    formato = image/png
    XMin = -412208.172942018
    YMin =  4928258.28942967
    XMax = -411725.664200968
    YMax =  4928740.79817072
    AnchoPixels =  101
    AltoPixels =  101
    Transparente = No
    Descripción error:
    layers (AD.ADDRESSA) No soportada.]]>
   </ServiceException>
</ServiceExceptionReport>
```

Reply:
thanks @allyoucanmap